src/core/main.c

```diff
@@ -355,11 +368,9 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
 
         } else if (streq(key, "systemd.confirm_spawn") && value) {
 
-                r = parse_boolean(value);
+                r = parse_confirm_spawn(value, &arg_confirm_spawn);
```

Comment:
normally it doesn't.
free_and_strdup() is used in parse_confirm_spawn() (as suggested by @keszybz).